bundlewrap/items/zfs_dataset.py

```diff
@@ -3,137 +3,154 @@
 from bundlewrap.exceptions import BundleError
 from bundlewrap.items import Item
 from bundlewrap.utils.text import mark_for_translation as _
+from bundlewrap.utils import cached_property
 
 
 class ZFSDataset(Item):
     """
-    Creates ZFS datasets and manages their options.
+    Creates ZFS datasets and manages their properties.
     """
     BUNDLE_ATTRIBUTE_NAME = "zfs_datasets"
     REJECT_UNKNOWN_ATTRIBUTES = False
     ITEM_TYPE_NAME = "zfs_dataset"
+    # for defaults which should be different from 'zfs inherit -S' behaviour
+    PROPERTY_DEFAULTS = {
+        'mountpoint': 'none',
+    }
 
     def __repr__(self):
-        return f"<ZFSDataset name:{self.name} {' '.join(f'{k}:{v}' for k,v in self.attributes.items())}>"
-
-    def __create(self, path, options):
-        option_list = []
-        for option, value in sorted(options.items()):
-            # We must exclude the 'mounted' property here because it's a
-            # read-only "informational" property.
-            if option != 'mounted' and value is not None:
-                option_list.append("-o {}={}".format(quote(option), quote(value)))
-        option_args = " ".join(option_list)
-
-        self.run(
-            "zfs create {} {}".format(
-                option_args,
-                quote(path),
-            ),
-            may_fail=True,
-        )
+        return f"<ZFSDataset name:{self.name} {' '.join(f'{k}:{v}' for k,v in self.__item_properties.items())}>"
```

Comment:
`__repr__()` should not need to run anything on the node. Calling `__item_properties()` will try to get the properties from the node itself.